language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/semantictokens/SemanticTokensVisitor.java

```diff
@@ -86,23 +87,32 @@
     // Sorted tree of semantic tokens
     private final Set<SemanticToken> semanticTokens;
     private final SemanticTokensContext semanticTokensContext;
+    private final String path;
+    private final SemanticModel semanticModel;
+
+    private Document document;
 
     public SemanticTokensVisitor(SemanticTokensContext semanticTokensContext) {
         this.semanticTokens = new TreeSet<>(SemanticToken.semanticTokenComparator);
         this.semanticTokensContext = semanticTokensContext;
+        Optional<String> optionalPath =
+                this.semanticTokensContext.workspace().relativePath(this.semanticTokensContext.filePath());
+        this.path = optionalPath.orElse(null);
+        Optional<SemanticModel> optionalSemanticModel = this.semanticTokensContext.currentSemanticModel();
+        this.semanticModel = optionalSemanticModel.orElse(null);
     }
 
     /**
      * Collects semantic tokens while traversing the semantic tress and returns the processed list of semantic tokens
      * for highlighting.
      *
-     * @param node Root node
+     * @param document Project document
      * @return {@link SemanticTokens}
      */
-    public SemanticTokens getSemanticTokens(Node node) {
+    public SemanticTokens getSemanticTokens(Document document) {
         List<Integer> data = new ArrayList<>();
-        visitSyntaxNode(node);
-
+        this.document = document;
+        visitSyntaxNode(document.syntaxTree().rootNode());
```

Comment:
> Why do we need to have the document as an argument here? Since the semantic token implementation is bound to a document that particular information can be captured from the operation context. So let's avoid passing the document here.

Passed the document since we have already used the document validation at SemanticTokensUtils.java and so we already have the document instance. And, accessing the document again consumes a significant time.

Reply:
Please use the `currentDocument` API. 
https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/AbstractDocumentServiceContext.java#L195

Reply:
Updated.